src/custom/state/lists/hooks/hooksMod.ts

```diff
@@ -194,3 +201,48 @@ export function useIsListActive(url: string): boolean {
   const activeListUrls = useActiveListUrls()
   return Boolean(activeListUrls?.includes(url))
 }
+
+export function useGpUnsupportedTokens(): UnsupportedToken | null {
+  const { chainId } = useActiveWeb3React()
+  return useSelector<AppState, AppState['lists'][ChainId]['gpUnsupportedTokens'] | null>(state =>
+    chainId ? state.lists[chainId].gpUnsupportedTokens : null
+  )
+}
+
+export function useAddGpUnsupportedToken() {
+  const dispatch = useDispatch()
+
+  return useCallback((params: AddGpUnsupportedTokenParams) => dispatch(addGpUnsupportedToken(params)), [dispatch])
+}
+
+export function useRemoveGpUnsupportedToken() {
+  const dispatch = useDispatch()
+
+  return useCallback((params: RemoveGpUnsupportedTokenParams) => dispatch(removeGpUnsupportedToken(params)), [dispatch])
+}
+
+export function useIsUnsupportedToken() {
+  const { chainId } = useActiveWeb3React()
+  const allUnsupportedTokens = useUnsupportedTokenList()
+  const gpUnsupportedTokens = useGpUnsupportedTokens()
+
+  return useCallback(
+    (address?: string) => {
+      // map unsupported token addresses by chainId to it's address in lower case
+      const unsupportedTokensList = chainId
+        ? Object.keys(allUnsupportedTokens[chainId]).map(address => address.toLowerCase())
+        : []
+      // map unsupported GP tokens by chainId to it's address in lower case
+      const gpUnsupportedTokensList =
+        chainId && gpUnsupportedTokens ? Object.keys(gpUnsupportedTokens).map(address => address.toLowerCase()) : []
+
+      // combine the lists to prepare for the Set init
+      const combined = unsupportedTokensList.concat(gpUnsupportedTokensList)
+      const unsupportedTokensMap = new Set(combined)
```

Comment:
Any way to keep a set of unsupported tokens in memory instead of building it every time the hook is called?

Reply:
it's in the callback so it's kinda cached already. but yeah i see what u mean. any suggestions?

Reply:
Don't know how much boilerplate is needed, but I'm thinking to build the set once and store it on the context.
On every add/remove unsupported tokens, update the set.